src/Windows/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -24,15 +26,28 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
 
         private RFormatOptions _formatOptions = new RFormatOptions();
 
-        public REditorSettings(ICoreShell coreShell): base(coreShell, RContentTypeDefinition.ContentType) { }
-        public REditorSettings(IEditorSettingsStorage storage) : base(storage) { }
+        public REditorSettings(ICoreShell coreShell) : base(coreShell, RContentTypeDefinition.ContentType) {
+            CreateLintOptions();
+        }
+
+        public REditorSettings(IEditorSettingsStorage storage) : base(storage) {
+            CreateLintOptions();
+        }
 
         public override void ResetSettings() {
             _formatOptions = new RFormatOptions();
-            LintOptions = new LintOptions();
+            CreateLintOptions();
             base.ResetSettings();
         }
 
+        private void CreateLintOptions() {
+            LintOptions = new LintOptions();
+            LintOptions.PropertyChanged += _OnLintOptionsPropertyChanged;
```

Comment:
I didn't want to use that storage since we need to clean it up. The problem is that in VS setting storage is a bit of a mess. Editor settings (at least some of them) have to use language storage since things like tab/space/indent/enable completion are provided by VS and have to be read from specific VS storage. However, options like Lint are stored in separately via DialogPage. Now, connections are stored in the roaming store... 

At some point I'd like to reduce language storage use (aka WritableStorage) to only settings that are provided/controlled by VS editor tab.